R/fitNullModel.R

```diff
@@ -17,25 +17,36 @@ setMethod("fitNullModel",
                    verbose = TRUE) {
 
               if (is.data.table(x)) x <- as.data.frame(x)
-              
-              desmat <- createDesignMatrix(x, outcome, covars, group.var)
 
+              desmat <- createDesignMatrix(x, outcome, covars, group.var)
               # if there was missing data, need to subset cov.mat
               if (!is.null(cov.mat)) {
                   .checkRownames(cov.mat, x)
                   if (nrow(desmat$X) < nrow(x)) {
                       ind <- which(rownames(x) %in% rownames(desmat$X))
                       cov.mat <- .covMatSubset(cov.mat, ind)
                   }
+                  cov.mat <-  .setCovMatNames(cov.mat)
               }
-              
-              .fitNullModel(y=desmat$y, X=desmat$X, covMatList=cov.mat,
+
+
+              out <- .fitNullModel(y=desmat$y, X=desmat$X, covMatList=cov.mat,
                             group.idx=desmat$group.idx, family=family,
                             start=start, AIREML.tol=AIREML.tol,
                             max.iter=max.iter, EM.iter=EM.iter,
                             drop.zeros=drop.zeros,
                             return.small=return.small,
                             verbose=verbose)
+              rownames(out$fit) <- rownames(desmat$y)
```

Comment:
What if we just assigned this to a column called `sample.id` here instead of the rownames? Sorry if I'm forgetting a reason why that doesn't work.

Reply:
See above, this was intentional behavior for data frames.